pulse2percept/fast_retina.pyx

```diff
@@ -0,0 +1,56 @@
+import cython
+cimport numpy as np
+import numpy as np
+
+
+cdef extern from "math.h":
+    cpdef float expf(float x)
+
+
+cdef inline float expit(float x):
+    return 1.0 / (1.0 + expf(x))
+
+cdef inline float float_max(float a, float b):
+    return a if a >= b else b
+
+DTYPE = np.float
+ctypedef np.float_t DTYPE_t
+
+
+def nanduri2012_model_cascade(np.ndarray[DTYPE_t, ndim=1] stimdata,
```

Comment:
I believe that both the input to this function (`stimdata`) as well as the output (`out_R4`) can be implemented as memory views. You might also get something out of refactoring the intermediate list (`tmp_R4a`) as a memory view into an array. Jake's excellent blogpost on the topic (https://jakevdp.github.io/blog/2012/08/08/memoryview-benchmarks/) explains why: memory views are designed to reduce overhead in reference counting when slicing into arrays (IIUC).